components/merkledb/src/validation.rs

```diff
@@ -1,7 +1,18 @@
 //! Validation helpers for index names.
 
+use core::fmt;
+
 /// Validate index name.
 pub fn is_valid_index_name<S: AsRef<str>>(name: S) -> bool {
+    name.as_ref()
+        .as_bytes()
+        .iter()
+        .copied()
+        .all(|c| is_allowed_latin1_char(c) || c == 46)
+}
+
+/// Validate index name prefix, it shouldn't contain the dot.
+pub fn is_valid_prefix<S: AsRef<str>>(name: S) -> bool {
```

Comment:
AFAICS, this is used not only for prefix. Maybe something like `is_valid_index_name_component` here and `is_valid_index_full_name` instead of `is_valid_index_name`?